massa-execution-worker/src/context.rs

```diff
@@ -459,6 +459,16 @@ impl ExecutionContext {
 
     /// Creates a new smart contract address with initial bytecode, and returns this address
     pub fn create_new_sc_address(&mut self, bytecode: Bytecode) -> Result<Address, ExecutionError> {
+        // pay the Cranelift compilation cost
+        let cost = self.config.gas_costs.cl_compilation_cost;
+        if let Some(remaining) = self.max_gas.checked_sub(cost) {
```

Comment:
If you wonder why the costs deduction have been placed where they are, the though process was (in no specific order) that they must be in a place where:
* they are in lowest function possible
* its logical for them to located in
* it's hard for a developper to make a mistake leading to erroneous cost handling
* a developper does not have to constantly think about them when making a change to execution

Reply:
This is outdated as I realised just now that the context `max_gas` value actually never held anything. Moved the CL compilation payment back to ABI costs and will update the doc accordingly. The statement still holds for the cache payments.

Reply:
I am not happy with the fact that CL compilation are still paid in ABI costs (lack of better alternatives), this is something that we will have to remember when making modifications.